src/avtp_vsf_stream.c

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (c) 2025, Collabora Ltd
+ * Olivier Crete <olivier.crete@collabora.com>
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *    * Redistributions of source code must retain the above copyright notice,
+ *      this list of conditions and the following disclaimer.
+ *    * Redistributions in binary form must reproduce the above copyright
+ *      notice, this list of conditions and the following disclaimer in the
+ *      documentation and/or other materials provided with the distribution.
+ *    * Neither the name of Intel Corporation nor the names of its contributors
+ *      may be used to endorse or promote products derived from this software
+ *      without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <arpa/inet.h>
+#include <endian.h>
+#include <string.h>
+
+#include "avtp.h"
+#include "avtp_vsf_stream.h"
+#include "avtp_stream.h"
+#include "util.h"
+
+#define SHIFT_VENDOR_ID_1      (31 - 31)
+#define SHIFT_VENDOR_ID_2      (31 - 31)
+
+#define MASK_VENDOR_ID_1      (BITMASK(32) << SHIFT_VENDOR_ID_1)
+#define MASK_VENDOR_ID_2      (BITMASK(16) << SHIFT_VENDOR_ID_2)
+
+static int get_vendor_id(const struct avtp_stream_pdu *pdu, uint64_t *val)
+{
+        uint32_t bitmap1, bitmap2;
+        uint64_t val1, val2;
+
+        bitmap1 = ntohl(pdu->format_specific);
+        bitmap2 = ntohl(pdu->packet_info);
+
+        val1 = BITMAP_GET_VALUE(bitmap1, MASK_VENDOR_ID_1, SHIFT_VENDOR_ID_1);
```

Comment:
I like the code consistency, but this `val1` doesn't need the bitmap step, right? We just hope the compiler optmizes it away?

Reply:
It doesn't, I left it for consistency, but I'm happy removing it.